php/class-coauthors-wp-list-table.php

```diff
@@ -94,18 +96,23 @@ function prepare_items() {
 			$this->active_filter = 'show-all';
 		}
 
-		switch ( $this->active_filter ) {
-			case 'with-linked-account':
-			case 'without-linked-account':
-				$args['meta_key'] = $coauthors_plus->guest_authors->get_post_meta_key( 'linked_account' );
-				if ( 'with-linked-account' == $this->active_filter ) {
-					$args['meta_compare'] = '!=';
-				} else {
-					$args['meta_compare'] = '=';
-				}
-				$args['meta_value'] = '0';
-				break;
-		}
+        $key = $coauthors_plus->guest_authors->get_post_meta_key( 'linked_account' );
+        switch ( $this->active_filter ) {
+            case 'with-linked-account':
+                $args['meta_query'] = array(
+                    'relation' => 'AND',
+                    array( 'key' => $key, 'compare' => 'EXISTS' ),
```

Comment:
What's the reasoning for this meta query check if we already have the one below where it checks for a key that does not equate to an empty value?

Reply:
Good catch, @rebeccahum. There's no reason having `array( 'key' => $key, 'compare' => 'EXISTS' )` in this case. I just updated that statement and removed `'relation' => 'AND'` as well, as that's no longer needed.